src/FileCache.php

```diff
@@ -135,27 +172,43 @@ protected function setValue($key, $value, $ttl): bool
         $this->gc();
         $cacheFile = $this->getCacheFile($key);
         if ($this->directoryLevel > 0) {
-            @FileHelper::createDirectory(\dirname($cacheFile), $this->dirMode, true);
+            try {
+                @FileHelper::createDirectory(\dirname($cacheFile), $this->dirMode, true);
+            } catch (\yii\exceptions\Exception $error) {
```

Comment:
Why would it throw an exception there if `@` is used?

Reply:
Adding "@" does not suppress exceptions. It only suppress side-effects of built-in functions. 
I am not sure about "yiisoft/cache" error handling policy (whether error-handler is registered to convert any warning to exceptions). 
For this particular case, I discovered that ::createDirectory can possibly throw an exception, which bubbling is unwanted in this particular case, and decided to wrap it with try/catch.

Reply:
In which case it may throw an exception? Is it a good idea to ignore this case?

Reply:
I am not sure, whether we want cache functionality to throw exceptions outside.